<h1> Prospects for Opening a Business in San Francisco </h1>

<h2> The Objective </h2>

San Francisco is a notoriously expensive location for individuals and businesses alike. Median rent prices were \\$3,500 a month in January 2020 and median home price was \\$1.3 million, far above the national medians of \\$1,200 and \\$285,000, respectively (source: Zillow for rent price, Trullia for housing price). Business Insider reports that, to buy a home in San Francisco, one would need to earn \\$303,000 a year.

San Francisco's high cost of living is due to its proximity to Silicon Valley. As a result, the high expenses have not discouraged tech employees from living in or in areas close to San Francisco. With this in mind, what are the business prospects for a company looking to locate itself in this expensive city by the bay? Although San Francisco is a relatively compact city compared to others in America, there are still 27 zip codes in San Francisco County (which exclusively covers the city of San Francisco and very close communities in the bay).

The goal of this project is to analyze the various neighborhoods in San Francisco to see what types of venues are present in order to assess which neighborhood would be most ideal for starting a new business. The Foursquare API will be used to gather information about theses venues. The justification for using venue data is that if there are a larger number of businesses, especially those that are doing well in a neighborhood, the greater the chance that there is a thriving neighborhood with prospective clients interested in becoming customers at a new, service-based business. As such, our <b> main audience </b> for this report will be prospective <b>service-based</b> businesses such as restaurants, coffee shops, or retail stores. While the presence of similar venues may signal increased competition, it also signals greater potential for customers as well. 

<h2> The Data for Analysis </h2>

Data will come from:
* [Foursquare API](https://foursquare.com/city-guide), where we will gather venue data about neighhorhoods
* The neighborhoods are defined by [the list of zip codes from zipcodemaps.com](https://www.zipdatamaps.com/san-francisco-ca-county-zipcodes), specifically those that are listed in the non-unique section of the website and have population numbers. The site also provides GPS coordinates for each neighborhood

Foursquare will provide not just the top 100 venues in these neighborhoods, but also give a sense of the quality of venues (through the tips system) in these neighborhoods. The quality of these venues suggested by the tips system will allow us to compile an overall score based on the number and quality of venues that will be used to compare the neighborhoods and their potential as new business locations. Maps visualizing the location of the top venues in each neighborhood will also be provided as a means of assessing the proximity of these venues to each other. This density of  venues can also be crucial for analysis, as closer proximity venues suggests more potential customers in each location.

<h2> Methodology </h2>

The steps for conducting this study were as follows:

- Import the list of zip codes and the longitude and latitude coordinates for each zip code from zipcodemaps.com

The [San Francisco link](https://www.zipdatamaps.com/san-francisco-ca-county-zipcodes) has all the zip codes listed for San Francisco county, which covers the city and a few limited surrounding areas. We dropped any zip code that did not have population values associated with it. Each zip code link from the website also includes longitude and latitude coordinates as well.

<b>Note:</b> We chose to use zip codes instead of neighborhoods in San Francisco due to the fact that San Francisco's neighborhoods are not well-defined. There are more than [110+ neighborhoods in San Francisco](https://en.wikipedia.org/wiki/List_of_neighborhoods_in_San_Francisco), many overlapping and redundant, making analysis with neighborhoods difficult. In contrast, since zip codes are defined by distance from a geographical center, the areas defined by zip codes are distinct and therefore easier to analyze.

For reference, the zip code database with coordinates is below:

In [5]:
df

index  Neighborhood  Population     Latitude      Longitude
0       0         94102       31176  37.77938843  -122.41828156
1       1         94103       27170  37.77603912  -122.41335297
2       2         94104         406  37.79093170  -122.40177917
3       3         94105        5846  37.78956985  -122.39475250
4       4         94107       26599  37.76816177  -122.39868164
5       6         94108       13768  37.79093170  -122.40892029
6       7         94109       55984  37.79748917  -122.42076874
7       8         94110       69333  37.74868011  -122.41504669
8       9         94111        3713  37.79978180  -122.39974213
9      10         94112       79407  37.72204971  -122.44474030
10     11         94114       31124  37.75825119  -122.43892670
11     12         94115       33021  37.78593826  -122.43714142
12     13         94116       43698  37.74367142  -122.48320770
13     14         94117       39169  37.76908112  -122.44291687
14     15         94118       38319  37.77867126  -122.46066284
15     16         94121       41203  37.78094864  -122.49443054
16     17         94122       56023  37.76182175  -122.48413086
17     18         94123       23088  37.80118179  -122.43576050
18     19         94124       33996  37.73096085  -122.38255310
19     20         94127       19289  37.73429871  -122.45664978
20     21         94129        3183  37.79895020  -122.46556854
21     22         94130        2880  37.81990051  -122.36789703
22     23         94131       26881  37.74570847  -122.44328308
23     24         94132       28129  37.72285080  -122.48438263
24     25         94133       26237  37.80403900  -122.41079712
25     26         94134       40798  37.72174072  -122.41040039
26     27         94158        4792  37.77244949  -122.39166260

- Getting a list of top venues from the Foursquare API

We used the Foursquare API to get a list of the top venues in each zip code. We selected for the top 100 venues in each zip code and then performed a manual analysis of the top five venues in each zip code. A cursory glance of the data seemed to indicate that food and drink venues such as restaurants, bars, and coffee shops are extremely common in the city.

- KMeans Clustering and created a map with Folium to assess the clusters

As mentioned, we used KMeans Clustering to group the venues into clusters and plotted the points onto a map of San Francisco using Folium. In analyzing the clusters, it appears that our manual analysis and hunch was correct, that food and drink venues tend to be very common in San Francisco. 

<h2> Results and Discussion </h2>

The findings seem to indicate that San Francisco has a large number of food and drink venues, which are clustered in the northeastern part of the city. This appears to indicate that this region of the city would be the best for a new business to be located, which is perhaps unsurprising as this is the location of San Francisco's Financial District and Downtown. 

The recommendation seems to be clear given the limit scope of our analysis. The best place to locate a new business would be in the northeastern part of the city, roughly corresponding to zip codes 94102, 94103, 94108, and 94109. As mentioned earlier, these zip codes roughly correspond to San Francisco's Financial District and Downtown, which are filled with corporate headquarters and existing service businesses, making it ideal for a new service-based business to be located in San Francisco.

<h2>Conclusion</h2>

The findings themselves in this project are perhaps not surprising, as it suggests that the best location for a new business would be in relatively wealthy and densely crowded parts of San Francisco, the Financial District and Downtown. With that being said, it is necessary also to consider the cost of real estate in these locations, a consideration that was not taken into account in this particular report. While it may be too expensive for a brand new business to locate itself within the confines of these zip codes itself, it is perhaps prudent to consider locating a business on the outer edges of these zip codes in order to benefit from a reasonably high level of customers and customer incomes while benefiting from somewhat lower cost of entry.